pkg/yamlpath/filter.go

```diff
@@ -130,9 +130,11 @@ func pathFilterScanner(n *filterNode) filterScanner {
 	}
 	return func(node, root *yaml.Node) []string {
 		if at {
-			return values(path.Find(node))
+			v, _ := path.Find(node)
+			return values(v)
 		}
-		return values(path.Find(root))
+		v, _ := path.Find(root)
+		return values(v)
```

Comment:
consider this alternative:

```
func values(nodes []*yaml.Node, err error) []string {
	if err != nil {
		panic(fmt.Errorf("the caller just can't fail / but it did; panic you should / and run for your life: %w", err))
	}
	// or ignore the err if you prefer.
...
}
```

now you can keep this call site as:

```
return values(path.Find(node))
```

and whenever you decide that `Find` now can in fact return errors, you change the signature of `value()` and have the compiler tell you which call-sites to update (otherwise it's easy to forget to update these `v, _` and percolate the error handling).


(working example: https://play.golang.org/p/bOub4PmZdHK)

Reply:
Yes, that'll dent the code coverage stats a bit, but who cares. Adopting...